api/censuses.go

```diff
@@ -231,6 +231,52 @@ func (capi *census3API) createAndPublishCensus(req *CreateCensusRequest, qID str
 	return newCensusID, nil
 }
 
+// transformCensus method transforms the census holders using the configured
+// providers and returns the final accuracy of the census. If the census is
+// anonymous, it rounds the balances and returns the final accuracy. If the
+// census is for farcaster, it uses the farcaster provider to transform the
+// holders. If none of the previous conditions are met, it returns the same
+// holders and an accuracy of 100%.
+func (capi *census3API) transformCensus(holders map[common.Address]*big.Int,
+	anonymous, farcaster bool,
+) (map[common.Address]*big.Int, float64, error) {
+	// if the census is anonymous, round the balances and return the final accuracy
+	if anonymous {
+		// Round census balances using the default configuration
+		censusParticipants := []*roundedcensus.Participant{}
+		for address, balance := range holders {
+			censusParticipants = append(censusParticipants, &roundedcensus.Participant{
+				Address: address.String(),
+				Balance: balance,
+			})
+		}
+		res, accuracy, err := roundedcensus.GroupAndRoundCensus(censusParticipants, roundedcensus.DefaultGroupsConfig)
+		if err != nil {
+			return nil, 0, err
+		}
+		// update the census holders with the rounded balances and the final accuracy
+		finalHolders := map[common.Address]*big.Int{}
+		for _, participant := range res {
+			finalHolders[common.HexToAddress(participant.Address)] = participant.Balance
+		}
+		return finalHolders, accuracy, nil
+	}
+	// // if farcaster is enabled, use the farcaster provider to transform the holders
+	// if farcaster {
+	// 	provider, ok := capi.holderProviders[providers.CONTRACT_TYPE_FARCASTER]
+	// 	if !ok {
+	// 		return nil, 0, fmt.Errorf("farcaster provider not configured")
+	// 	}
+	// 	farcasterHolders, err := provider.CensusKeys(holders)
+	// 	if err != nil {
+	// 		return nil, 0, err
+	// 	}
+	// 	return farcasterHolders, 100, nil
+	// }
+	// by default, return the same holders and an accuracy of 100%
```

Comment:
this code is only required for token gated use case, right?

Reply:
yes right